presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateCountDistinctAggregation.java

```diff
@@ -120,6 +121,29 @@ public static void input(
         state.addMemoryUsage(hll.estimatedInMemorySize());
     }
 
+    @InputFunction
+    public static void input(BooleanDistinctState state,
+            @SqlType(StandardTypes.BOOLEAN) boolean value,
+            @SqlType(StandardTypes.DOUBLE) double maxStandardError)
+    {
+        //States:
+        // 0 -> no values seen
+        // 1 -> true seen
+        // 2 -> false seen
+        // 3 -> both true and false seen
+        switch (state.getInt()) {
```

Comment:
What about use bit operations:

```
int ordinal = (value ? 1 : 0);
state.setInt(state.getInt || (1 << ordinal));
```

Reply:
@wenleix Changed the implementation to comment by @findepi of form
   `     state.setByte((byte) (state.getByte() | (value ? 1 : 2)));`
Please let me know if that makes sense.